src/main/java/de/uos/informatik/ko/rcp/geneticAlgo/GeneratePop.java

```diff
@@ -0,0 +1,101 @@
+package de.uos.informatik.ko.rcp.geneticAlgo;
+
+import java.lang.reflect.Array;
+import java.lang.Integer;
+import java.util.*;
+import de.uos.informatik.ko.rcp.Utils;
+import de.uos.informatik.ko.rcp.Instance;
+
+
+public class GeneratePop {
+
+    private GeneratePop() {}
+    //change to Hashset of ArrayLists
+    public static HashSet<ArrayList<Integer>> generatePop(Instance instance, Integer pop_size,Random myGenerator){
+        // all activities without predecessor are possible candidates
+        HashSet<ArrayList<Integer>> results = new HashSet<ArrayList<Integer>>();
+        HashMap<Integer,HashSet<Integer>> mypredecessors;
+        HashSet<Integer> candidates = new HashSet<Integer>();
+        //scheduled activities
+        ArrayList<Integer> scheduled = new ArrayList<Integer>();
+        for (int i=0; i< pop_size*5; i++) {
+            mypredecessors = Utils.buildPredecessorMap(instance);
+            for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                if (entry.getValue().isEmpty() && (!(scheduled.contains(entry.getKey())))) {
+                    candidates.add(entry.getKey());
+                }
+            }
+            Integer mycandidate = chooseone(candidates, myGenerator);
+            scheduled.add(mycandidate);
+            candidates.remove(mycandidate);
+            //delete scheduled activity from predecessor list
+            for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                entry.getValue().remove(mycandidate);
+            }
+            while (scheduled.size() < instance.n()) {
+                for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                    if (entry.getValue().isEmpty()&& (!(scheduled.contains(entry.getKey())))) {
+                        candidates.add(entry.getKey());
+                    }
+                }
+                Integer actcandidate = chooseone(candidates, myGenerator);
+                scheduled.add(actcandidate);
+                candidates.remove(actcandidate);
+                // delete scheduled activity from predecessor list
+                for (Map.Entry<Integer, HashSet<Integer>> entry : mypredecessors.entrySet()) {
+                    entry.getValue().remove(actcandidate);
+                }
+            }
+            ArrayList<Integer> resultsList = new ArrayList<Integer>();
+            for (Integer elem: scheduled){
+                resultsList.add(elem);
+            }
```

Comment:
We can use `ArrayList`'s ["copy constructor"](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ArrayList.html#%3Cinit%3E(java.util.Collection)):
```suggestion
            ArrayList<Integer> resultsList = new ArrayList<Integer>(scheduled);
```

Reply:
You’re right.